ArduCopter/mode_guided.cpp

```diff
@@ -436,29 +475,44 @@ void ModeGuided::pos_control_run()
     // set motors to full range
     motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
 
-    // run waypoint controller
-    copter.failsafe_terrain_set_status(wp_nav->update_wpnav());
+    bool do_avoid = false;
+    guided_vel_target_cms = pos_control->get_vel_target_cms();
+#if AC_AVOID_ENABLED
+    // limit the velocity for obstacle/fence avoidance
+    copter.avoid.adjust_velocity(guided_vel_target_cms, pos_control->get_pos_xy_p().kP(), pos_control->get_max_accel_xy_cmss(), pos_control->get_pos_z_p().kP(), pos_control->get_max_accel_z_cmss(), G_Dt);
+    do_avoid = copter.avoid.limits_active();
+#endif
+
+    // send position and velocity targets to position controller
+    guided_accel_target_cmss.zero();
+    if (do_avoid) {
+        pos_control->input_vel_accel_xy(guided_vel_target_cms, guided_accel_target_cmss);
+        pos_control->input_vel_accel_z(guided_vel_target_cms, guided_accel_target_cmss, false);
+    } else {
+        guided_vel_target_cms.zero();
+        pos_control->input_pos_vel_accel_xyz(guided_pos_target_cm);
```

Comment:
it's existing code but I wonder why AC_PosControl::input_pos_vel_accel_xyz is not called AC_PosControl::input_pos_xyz.